### ISSUE TITLE
Newline in Python file does not indent

### ISSUE THREAD

**Issue by failable** [zed-industries/zed#16288]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After a recent update, newline in Python file does not indent correctly.

Tested with empty user config `{}`:

<img width="652" alt="image" src="https://github.com/user-attachments/assets/cc8af462-afd9-4a33-a5a1-37a3a9e20099">

(Left: the user config; Right: Insert `{}` and press enter between the braces)

### Environment

Zed: v0.149.0 (Zed Preview)
OS: macOS 14.6.0
Memory: 32 GiB
Architecture: x86_64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by failable** [comment]
I don't see any specific information in the logs.

After some tests, it seems it's related to the `basedpyright` extension. 

- With an empty user setting and `basedpyright` not installed and fresh start of Zed, the issue does not exist.
- With an empty user setting and `basedpyright` installed, the issue exists.
- With an empty user setting and `basedpyright` uninstalled without restarting Zed, the issue exists.
- With an empty user setting and `basedpyright` uninstalled and restarting Zed, the issue is gone.

**Comment by notpeter** [comment]
Interesting. Wasn't familiar with https://github.com/m1guer/basedpyright-zed

I think you need to use their recommended settings which disable the built-in pyright:
```json

{
  "languages": {
    "Python": {
      "language_servers": ["basedpyright", "!pyright"]
  }
}
```

**Comment by failable** [comment]
Yes, I originally have configs like that. But for debugging I removed them and found the issue persist.
